MapleServer2/Database/Classes/DatabaseAccount.cs

```diff
@@ -0,0 +1,125 @@
+﻿using Maple2Storage.Types;
+using MapleServer2.Enums;
+using MapleServer2.Types;
+using Newtonsoft.Json;
+using SqlKata.Execution;
+
+namespace MapleServer2.Database.Classes
+{
+    public static class DatabaseAccount
+    {
+        public static long CreateAccount(Account account)
+        {
+            return DatabaseManager.QueryFactory.Query("Accounts").InsertGetId<long>(new
+            {
+                account.Username,
+                account.PasswordHash,
+                account.CreationTime,
+                account.LastLoginTime,
+                account.CharacterSlots,
+                Meret = account.Meret.Amount,
+                GameMeret = account.GameMeret.Amount,
+                EventMeret = account.EventMeret.Amount,
+            });
+        }
+
+        public static Account FindById(long id)
+        {
+            return ReadAccount(DatabaseManager.QueryFactory.Query("Accounts").Where("Accounts.Id", id)
+            .LeftJoin("Homes", "Homes.AccountId", "Accounts.Id")
+            .Select("Accounts.{*}", "Homes.Id as HomeId")
+            .FirstOrDefault());
+        }
+
+        public static Account FindByUsername(string username) => ReadAccount(DatabaseManager.QueryFactory.Query("Accounts").Where("Username", username).FirstOrDefault());
+
+        public static bool Authenticate(string username, string password, out Account account)
+        {
+            account = null;
+            Account dbAccount = FindByUsername(username);
+
+            if (BCrypt.Net.BCrypt.Verify(password, dbAccount.PasswordHash))
+            {
+                account = dbAccount;
+                return true;
+            }
+            return false;
+        }
+
+        internal static List<Player> FindAllByAccountId(long accountId)
+        {
+            List<Player> characters = new List<Player>();
+
+            IEnumerable<dynamic> result = DatabaseManager.QueryFactory.Query("Characters").Where("AccountId", accountId)
```

Comment:
Maybe querying the characters table should not happen in the `Account` file?

Reply:
oh 100% my bad, it was one of the first stuff I worked on